CanDIG/katsu
# DIG-1239: Donor explorer api

## What's new:
- Implemented DonorExplorerSchema, a new API that requires query service token.
- Allows filtering by primary site, treatment type, chemotherapy drug name, immunotherapy drug name, hormone therapy drug name, and cohorts.
- Returns a list of donor objects including program ID, submitter donor ID, submitter sample IDs, primary site, treatment type, and age at diagnosis.

### How to test:
- Local Testing:
  - Run tox.
  - Ingest a small dataset.
  - Access the Swagger UI and use the token `query` to test the API.
  - Expected outcome: Verify that the API returns a list of donor objects according to the specified filtering criteria.

- Docker Testing:
  - Pull this branch.
  - Execute the `test-integration`.
  - Expected outcome: Ensure that the integration tests pass successfully.
